src/test/java/seedu/address/model/person/GradeTest.java

```diff
@@ -0,0 +1,37 @@
+package seedu.address.model.person;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import seedu.address.testutil.Assert;
+
+public class GradeTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> new Grade(null));
+    }
+
+    @Test
+    public void constructor_invalidSubject_throwsIllegalArgumentException() {
+        String invalidSubject = "";
+        Assert.assertThrows(IllegalArgumentException.class, () -> new Grade(invalidSubject));
+    }
+
+    @Test
+    public void isValidGrade() {
```

Comment:
wrong naming syntax. remember the checkstyle. This is the same for the other tests of this nature,